packages/rtk-query-codegen-openapi/src/bin/cli.ts

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env node
+
+import program from 'commander';
+import { dirname, resolve } from 'path';
+import { generateEndpoints, parseConfig } from '../';
+
+let ts = false;
+try {
+  if (require.resolve('esbuild') && require.resolve('esbuild-runner')) {
+    require('esbuild-runner/register');
+  }
+  ts = true;
+} catch {}
+
+try {
+  if (!ts) {
+    if (require.resolve('typescript') && require.resolve('ts-node')) {
+      require('ts-node/register/transpile-only');
```

Comment:
> @phryneas No, there is a bunch of extra stacktrace after. Here is the whole thing:
> 
> ```shell
> $ node ./node_modules/@rtk-query/codegen-openapi/lib/bin/cli.js openapi.dev.config.ts
> (node:33479) UnhandledPromiseRejectionWarning: /Users/austinfelix/apps/finsense-ui/openapi.dev.config.ts:10
> export default config;
> ^^^^^^
> 
> SyntaxError: Unexpected token 'export'
>     at wrapSafe (internal/modules/cjs/loader.js:1001:16)
>     at Module._compile (internal/modules/cjs/loader.js:1049:27)
>     at Module.m._compile (/Users/austinfelix/apps/finsense-ui/node_modules/ts-node/src/index.ts:1365:23)
>     at Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
>     at Object.require.extensions.<computed> [as .ts] (/Users/austinfelix/apps/finsense-ui/node_modules/ts-node/src/index.ts:1368:12)
>     at Module.load (internal/modules/cjs/loader.js:950:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:790:14)
>     at Module.require (internal/modules/cjs/loader.js:974:19)
>     at require (internal/modules/cjs/helpers.js:92:18)
>     at run (/Users/austinfelix/apps/finsense-ui/node_modules/@rtk-query/codegen-openapi/src/bin/cli.ts:45:26)
> (Use `node --trace-warnings ...` to show where the warning was created)
> (node:33479) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> (node:33479) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> ```
> 
> I did switch to esr and it works perfectly with the original config file. However, it may just be a local development problem. I think most people will use npx as described in the docs once its released. You might see some create a script and run it with ts-node if they want to do something really custom.


So I believe we will have to add some config here so it uses a lower transpilation target/commonJS output. I'll have to experiment with that in the next few days.